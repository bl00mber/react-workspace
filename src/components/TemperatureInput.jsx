import React from 'react';
import PropTypes from 'prop-types';

const scaleNames = {
  c: 'Celsius',
  f: 'Fahrenheit',
};

class TemperatureInput extends React.Component {
  handleChange(e) {
    const { onTemperatureChange } = this.props;
    onTemperatureChange(e.target.value);
  }

  render() {
    const { scale, temperature } = this.props;
    return (
      <fieldset>
        <legend>
          Enter temperature in&nbsp;
          {scaleNames[scale]}
          :
        </legend>
        <input
          value={temperature}
          onChange={e => this.handleChange(e)}
        />
      </fieldset>
    );
  }
}

TemperatureInput.propTypes = {
  temperature: PropTypes.string.isRequired,
  scale: PropTypes.string.isRequired,
  onTemperatureChange: PropTypes.func.isRequired,
};

export default TemperatureInput;
