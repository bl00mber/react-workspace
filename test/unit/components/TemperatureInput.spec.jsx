import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import TemperatureInput from '@/components/TemperatureInput';

describe('TemperatureInput', () => {
  test('should match enzyme shallow snapshot', () => {
    const props = {
      temperature: '100',
      scale: 'c',
      onTemperatureChange: jest.fn(),
    };
    const wrapper = shallow(<TemperatureInput {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
