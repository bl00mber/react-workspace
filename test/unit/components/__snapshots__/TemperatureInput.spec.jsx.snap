// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TemperatureInput should match enzyme shallow snapshot 1`] = `
<fieldset>
  <legend>
    Enter temperature in 
    Celsius
    :
  </legend>
  <input
    onChange={[Function]}
    value="100"
  />
</fieldset>
`;
